components/locale/macros/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+
+fn get_value_from_token_stream(input: TokenStream) -> String {
+    let val = format!("{}", input);
+    if !val.starts_with('"') || !val.ends_with('"') {
+        panic!("Argument must be a string literal.");
+    }
+    let len = val.len();
+    (&val[1..len - 1]).to_string()
+}
+
+fn language_output(input: Option<u64>) -> String {
+    let val = if let Some(raw) = input {
+        format!("Some({})", raw)
+    } else {
+        "None".to_string()
+    };
+    format!("unsafe {{ icu_locale::subtags::Language::from_raw_unchecked({}) }}", val)
+}
+
+fn script_output(input: u32) -> String {
+    format!("unsafe {{ icu_locale::subtags::Script::from_raw_unchecked({}) }}", input)
+}
+
+fn region_output(input: u32) -> String {
+    format!("unsafe {{ icu_locale::subtags::Region::from_raw_unchecked({}) }}", input)
+}
+
+fn variant_output(input: u64) -> String {
+    format!("unsafe {{ icu_locale::subtags::Variant::from_raw_unchecked({}) }}", input)
+}
+
+#[proc_macro]
+pub fn language(input: TokenStream) -> TokenStream {
```

Comment:
Here and below: I think macros aren't namespaced.  `language!` seems awfully generic.  Consider `icu_language!`?  I did `icu_data_key!` for my macro in DataProvider.

Reply:
I don't see a reason to prefix the macro names with `icu_` - I don't see a real chance that someone will have imported multiple macros named `language` into the same module.

If you feel strongly about this, I'll open an issue.

Reply:
Filed https://github.com/unicode-org/icu4x/issues/253